docs/samples/ble/raspberry-esp32-xiaomi-sensor-htu21d.md

```diff
@@ -0,0 +1,99 @@
+
+Data is collected by program that is running on [Raspberry Pi](https://en.wikipedia.org/wiki/Raspberry_Pi). It pushes data to ThingsBoard server via MQTT protocol by using  [ThingsBoard MQTT client Python SDK](https://github.com/thingsboard/thingsboard-python-client-sdk) library. Data is visualized using built-in customizable dashboard. Demo that is running on Raspberry Pi is written in Python which is quite simple and easy to understand.
+
+## List of hardware
+* ESP32-PICO-KIT 
+* HTU21D Digital Relative Humidity Sensor
+* Breadboard
+* 4 female-to-male jumper wires
+* Raspberry Pi with Bluetooth
+* Xiaomi Smart Temperature & Humidity Sensor
+
+**_TODO:_ Add photos of hardware**
+
+
+## Hardware setup
+#### ESP32 Wiring scheme
+| ESP32 | HTU21D |
+| ----- | ------ |
+| 3V3   | VIN    |
+| GND   | GND    |
+| 18    | SDA    |
+| 19    | SCL    |
+
+The following picture summarizes the connection of ESP32 and HTU21D:
+**_TODO:_ Add final picture of setup**
+
+#### Raspberry Pi
+In this setup we use Raspberry Pi 2 Model B V1.1 with Raspbian GNU/Linux 9.8 (stretch) installed on it. Raspbian is a free operating system based on Debian optimized for the Raspberry Pi hardware. This model does't support Bluetooth, so it needs a usb dongle. We use Grand-X BT40G.
+Make sure your dongle is recognized by system. You can check it with the following command:
+```
+$> lsusb
+...
+Cambridge Silicon Radio, Ltd Bluetooth Dongle (HCI mode)
+...
+
+```
+
+## Preparing ESP32
+You need ESP-IDF (Espressif IoT Development Framework) to program your ESP32.
+It is the official development framework for the ESP32 chip.
+Install it with the following instructions:
+```
+mkdir ~/esp
+cd ~/esp
+git clone --recursive https://github.com/espressif/esp-idf.git
+git checkout 3bf56cdd1
+```
+Set up IDF_PATH before running any other command:
+```
+export IDF_PATH=~/esp/esp-idf
+```
+The python packages required by ESP-IDF are located in IDF_PATH/requirements.txt. You can install them by running:
+```
+python -m pip install --user -r $IDF_PATH/requirements.txt
+```
+##### Note
+Please check the version of the Python interpreter that you will be using with ESP-IDF. For this, run the command `python --version` and depending on the result, you might want to use python2, python2.7 or similar instead of just python, e.g.
+
+See [this](https://docs.espressif.com/projects/esp-idf/en/latest/get-started/index.html) guide for more info.
+
+Now you can flash your ESP:
+```
+git clone name_of_repo_with_firmware
+cd name_of_repo_with_firmware
+make flash
+```
+
+
+## Preparing Raspberry Pi
+This demo requires at least **python3.4**. Install the following packages:
```

Comment:
Add `Login to your Raspberry PI using SSH` part of the sentence, to remind the user one more time about our setup